plugins/modules/redfish_powerstate.py

```diff
@@ -192,13 +270,51 @@ def is_change_applicable_for_power_state(current_power_state, apply_power_state)
 
 
 def is_valid_reset_type(reset_type, allowable_enum, module):
+    reset_type_param = module.params.get('reset_type')
     if reset_type not in allowable_enum:
         res_list = re.findall('[A-Z][^A-Z]*', reset_type)
         lw_reset_type = " ".join([word.lower() for word in res_list])
-        error_msg = "The target device does not support a" \
-                    " {0} operation.The acceptable values for device reset types" \
-                    " are {1}.".format(lw_reset_type, ", ".join(allowable_enum))
-        module.fail_json(msg=error_msg)
+        if reset_type_param:
+            error_msg = INVALID_RESET_TYPE.format(lw_reset_type, ", ".join(allowable_enum))
+        else:
+            error_msg = INVALID_RESET_TYPE_OEM.format(option=reset_type,
+                                                      supported_oem_reset_type_values=", ".join(allowable_enum))
+        module.exit_json(msg=error_msg, failed=True)
+
+
+def is_valid_final_pwr_state(final_pwr_state, allowed_final_pwr_state, module):
+    if final_pwr_state.capitalize() not in allowed_final_pwr_state:
+        error_msg = INVALID_PWR_ERROR.format(final_pwr_state, ", ".join(allowed_final_pwr_state))
+        module.exit_json(msg=error_msg, failed=True)
+
+
+def is_valid_vendor(redfish_session_obj, module, vendor):
+    # if vendor is None:
```

Comment:
Remove this commented line , if not required.

Reply:
Done